libcontainer/rootfs_linux.go

```diff
@@ -107,7 +116,10 @@ func mountToRootfs(m *configs.Mount, rootfs, mountLabel string) error {
 		if err := os.MkdirAll(dest, 0755); err != nil {
 			return err
 		}
-		return syscall.Mount(m.Source, dest, m.Device, uintptr(m.Flags), "")
+		if err := syscall.Mount(m.Source, dest, m.Device, uintptr(m.Flags), ""); err != nil {
+			return err
+		}
+		return setMountPropagation(dest, syscall.MS_PRIVATE|syscall.MS_REC)
```

Comment:
Ok, will copy some of the information from commit to source code as comments.

Right now all mounts inside container are private. But as we are now opening the possibility that some of the mounts can be shared/slave we want to define very well what will be shared/slave.  So only mounts of type "bind" can be shared/slave and rest will be private. That's why I am forcing all other type of mounts "proc,sysfs,mqueue, tmpfs, devpts" etc to be private. That way we know what kind of mounts can have property other then private.